Automattic/jetpack
# Build: Add yarn resolutions for vulnerable packages

Fixes GitHub security vulnerability warnings by defining yarn `resolutions` for vulnerable packages. From my understanding, this redefines dependency locks in the dependency tree by pulling a fixed version of the package. Since vulnerabilities are fixed in a bunch of different versions of these packages, yarn will pull the latest minor version, meaning it's relatively safe PR, and should not break ~much~ anything. 

It would be nice to merge it at the beginning of the 8.5 cycle, so it would intensively be tested before the release. 

yarn `resolutions` docs: https://classic.yarnpkg.com/en/docs/selective-version-resolutions/

Beware! Side effects of this PR are unknown 😬 

`minimist`-affected dependencies:

```
[ 'gulp-sass',
  '@wordpress/dependency-extraction-webpack-plugin',
  'css-loader',
  '@automattic/calypso-build',
  'eslint-loader',
  'sass-loader',
  'style-loader',
  'url-loader',
  'babel-loader',
  'copy-webpack-plugin',
  'webpack',
  'webpack-cli',
  'gulp',
  'jest',
  'babel-jest',
  'danger',
  '@babel/core',
  'jest-puppeteer',
  'node-wp-i18n',
  'node-sass',
  'eslint',
  'puppeteer',
  'gulp-rtlcss',
  'mapbox-gl' ]
```

`acorn`-affected dependencies:

```
[ 'jest',
  'jsdom',
  'eslint',
  '@wordpress/dependency-extraction-webpack-plugin',
  '@automattic/calypso-build',
  'webpack' ]
```

Before: `1135 vulnerabilities found - Packages audited: 1132809`
After: `19 vulnerabilities found - Packages audited: 1132809`

#### Changes proposed in this Pull Request:

* Add yarn resolutions for both `acorn` and `minimist`
* Fix 1116 vulnerabilities according to `yarn audit`

#### Testing instructions:






* Check Travis. All the builds should be green
* Play around with Jetpack dashboard and Jetpack blocks. Make sure they behave as expected
* Look for any JS errors

#### Proposed changelog entry for your changes:

* n/a
